pkg/auth/providers/ldap/ldap_client.go

```diff
@@ -73,6 +73,19 @@ func (p *ldapProvider) loginUser(credential *v32.BasicLogin, config *v3.LdapConf
 		fmt.Sprintf("(%v=%v)", ObjectClass, config.UserObjectClass),
 		operationalAttrList, nil)
 	opResult, err := lConn.Search(searchOpRequest)
+
+	// If no object is found, try again using SA
+	if err != nil && ldapv3.IsErrorWithCode(err, ldapv3.LDAPResultNoSuchObject) {
+		// Re-bind SA
+		err = ldap.AuthenticateServiceAccountUser(serviceAccountPassword, serviceAccountUserName, "", lConn)
+		if err != nil {
+			return v3.Principal{}, nil, err
```

Comment:
This is an instance of what @prachidamle was referring to. If you look below on L90 (in your branch) you can see that we wrap these errors in a httperror object. We do this so that when the error is returned to the user, they see a reasonable error code/message rather than something generic.

I would recommend doing a wrap error like the below does:

```go
return v3.Principal{}, nil, httperror.WrapAPIError(err, httperror.Unauthorized, "authentication failed")
```